### ISSUE TITLE
Infered mutation options has invalid `variables` type

### ISSUE THREAD

**Issue by LiniovasDovydas** [HuolalaTech/react-query-kit#47]
The following code has correct type in on success argument
```ts
const useAccountCreateMutation = createMutation<SeedAccountDto, SeedAccountRequest, Error>({
  mutationKey: ['account', 'create'],
  mutationFn: async (variables) =>
    ApiLayer.accounts.seedCreate(variables).then((res) => res.data.data!),
  onSuccess: (response, variables) => {
    console.log(variables);
  },
});

export { useAccountCreateMutation };
```

But when used as an option in hook its type becomes `Error`
![image](https://github.com/HuolalaTech/react-query-kit/assets/41839786/daa66b06-ef50-49d4-96b1-3cfb0206b600)

**Comment by liaoliao666** [comment]
Fixed in v3.1.3

**Comment by LiniovasDovydas** [comment]
Thank you, it works!
